File: design-patterns/java/chainofresponsibility/RateLimitHandler.java
public class RateLimitHandler extends BaseHandler {
    @Override
    public void handle(Request request) {
        if (request.requestCount >= 100) {
            System.out.println("RateLimitHandler: ❌ Rate limit exceeded.");
            return;
        }
        System.out.println("RateLimitHandler: ✅ Within rate limit.");
        forward(request);
    }
}

File: design-patterns/python/chainofresponsibility/rate_limit_handler.py
from .base_handler import BaseHandler
from .request import Request

class RateLimitHandler(BaseHandler):
    def handle(self, request: Request) -> None:
        if request.request_count > 100:
            print(f"Rate limit exceeded for user: {request.user}")
            return
        print(f"Rate limit check passed for user: {request.user}")
        self._forward(request) 